server/server.go

```diff
@@ -18,63 +18,79 @@ package server
 
 import (
 	"fmt"
-	"sync"
-	"time"
 
 	"metacontroller.io/controller/common"
 
 	"metacontroller.io/controller/decorator"
 	"metacontroller.io/options"
 
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
 	"metacontroller.io/apis/metacontroller/v1alpha1"
 	mcclientset "metacontroller.io/client/generated/clientset/internalclientset"
 	"metacontroller.io/controller/composite"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 
 	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
 )
 
-type controller interface {
```

Comment:
This interface is not needed anymore because `controller-runtime` already defines a `Reconciler` interface which is implemented by all controllers. Stopping the controllers managed by the manager is done automatically when the manager itself is stopped.